src/internal/streamstore/streamstore.go

```diff
@@ -12,14 +12,37 @@ import (
 	"github.com/alcionai/corso/src/pkg/path"
 )
 
-const (
-	// detailsItemName is the name of the stream used to store
-	// backup details
-	detailsItemName = "details"
-	// collectionPurposeDetails is used to indicate
-	// what the collection is being used for
-	collectionPurposeDetails = "details"
-)
+// ---------------------------------------------------------------------------
+// interfaces
+// ---------------------------------------------------------------------------
+
+// Streamer is the core interface for all types of data streamed to and
+// from the store.
+type Streamer interface {
+	Writer
+	Reader
+	Delete(context.Context, string) error
+}
+
+type Reader interface {
+	Read(context.Context, string, Unmarshaller, *fault.Bus) error
```

Comment:
I can't quite put my finger on why, but the fact that `Read` also performs the unmarshalling makes me uncomfortable. Maybe it's because it seems like it will make debugging unmarshall errors a bit harder (need to check whether it was an error unmarshalling the data or before trying to unmarshall the data). Maybe it's because it feels like it will make streaming data bit more difficult (i.e. does all the stream processing now have to be part of the `Unmarshaller` callback?)